service/src/test/java/bio/terra/cbas/dao/TestRunDao.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.cbas.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.cbas.dao.RunDao.RunsFilters;
+import bio.terra.cbas.dao.util.WhereClause;
+import bio.terra.cbas.models.CbasRunStatus;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+
+public class TestRunDao {
+
+  // Sort the NON_TERMINAL_STATES so that we can guarantee the order (based on enum ordination):
+  // UNKNOWN, QUEUED, INITIALIZING, RUNNING, PAUSED, CANCELING
+  private static final List<CbasRunStatus> sortedNonTerminalStatuses =
+      CbasRunStatus.NON_TERMINAL_STATES.stream().sorted().toList();
+
+  @Test
+  void emptyRunFilters() {
+    RunsFilters filters = RunsFilters.empty();
+    assertEquals(new WhereClause(List.of(), Map.of()), filters.buildWhereClause());
+  }
+
+  @Test
+  void emptyRunFilters2() {
+    RunsFilters filters = new RunsFilters(null, null);
+    assertEquals(new WhereClause(List.of(), Map.of()), filters.buildWhereClause());
+  }
+
+  @Test
+  void emptyRunFilters3() {
+    RunsFilters filters = new RunsFilters(null, List.of());
+    assertEquals(new WhereClause(List.of(), Map.of()), filters.buildWhereClause());
+  }
+
+  @Test
+  void runSetId() {
+    UUID uuid = UUID.randomUUID();
+    RunsFilters filters = new RunsFilters(uuid, null);
+    WhereClause actual = filters.buildWhereClause();
+    assertEquals("WHERE (run.run_set_id = :runSetId)", actual.toString());
+    assertEquals(Map.of("runSetId", uuid), actual.params());
+  }
+
+  @Test
+  void filterSortedNonTerminalStates() {
+
+    RunsFilters filters = new RunsFilters(null, sortedNonTerminalStatuses);
+    WhereClause actual = filters.buildWhereClause();
+    assertEquals(
+        "WHERE (run.status in (:status_0,:status_1,:status_2,:status_3,:status_4,:status_5))",
```

Comment:
So those numbers are _just_ placeholders in the postgres stored procedure (but on demain, so I guess it's an ad hoc stored procedure? ... 🤷 ), to be replaced by the real values when the query is run.

Eg if the query is `SELECT ... WHERE (status in :status_0, :status_1, :status_2)`
and out input parameters are:
```
status_0 -> RUNNING
status_1 -> COMPLETE
status_2 -> UNKNOWN
```

Then when postgres comes to actually run the query, it will interpolate the right values into the right places and get `SELECT ... WHERE (status in :RUNNING, COMPLETE, UNKNOWN)`.

As for "why do this indirect thing at all, why not just build the right query in the first place", it's basically all just there to protect against sql injection